cfme/login.py

```diff
@@ -110,7 +110,24 @@ def login(username, password, submit_method=_js_auth_fn):
         sel.sleep(1.0)
 
         logger.debug('Logging in as user %s' % username)
-        fill(form, {'username': username, 'password': password})
+        try:
+            fill(form, {'username': username, 'password': password})
+        except sel.InvalidElementStateException as e:
+            logger.warning("Got an error. Details follow.")
+            msg = str(e).lower()
+            if "element is read-only" in msg:
+                logger.warning("Got a read-only login form, will reload the browser.")
+                # Reload browser
+                quit()
+                ensure_browser_open()
+                sel.sleep(1.0)
+                sel.wait_for_ajax()
```

Comment:
Good point. I can bash something together when I have to write this again. :smile_cat: